simphony/testing/abc_check_engine.py

```diff
@@ -35,46 +45,48 @@ def create_dataset_items(self):
         """
 
     @abc.abstractmethod
-    def check_emtpy_dataset(self, ds):
-        """ Check if a dataset is empty
-        """
-
-    @abc.abstractmethod
     def check_instance_of_dataset(self, ds):
         """ Check if a dataset is instance of a class
         """
 
-    @abc.abstractmethod
-    def check_is_usable_dataset(self, ds, items):
-        """ Check if a dataset is usable
-        """
+    operation_mapping = {
+        'compare datasets': 'none',
+        'add item': 'none'}
+
+    def compare_operation(self, *args, **kwrds):
+        method = getattr(
+            sys.modules[__name__],
```

Comment:
I think this should be changed somehow.  

I understand that `create_dataset` returns either an ABCMesh, ABCParticles or ABCLattice and then you need to know which compare function to use (e.g. `compare_particles_datasets`...). 

 Is there another way that ParticlesCudsCheck, LatticeCudsCheck and MeshCudsCheck provide this information (e.g. an abstractmethod which they each implement to provide this information)?